src/VariantGenerator/SetupResult.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusImagePlugin\VariantGenerator;
+
+class SetupResult implements SetupResultInterface
+{
+    private string $generatorName;
+
+    /** @var array<string, string> */
+    private array $messages = [];
+
+    public function __construct(string $generatorName)
+    {
+        $this->generatorName = $generatorName;
+    }
+
+    public function getGeneratorName(): string
+    {
+        return $this->generatorName;
+    }
+
+    public function addMessage(string $variantName, string $message): void
```

Comment:
This data structure (`array<variant name, string>`) is very specific. I would expect `list<string>`. I guess it ties very much into the definition list from the console? ;)

I feel it should be more generic.

Reply:
Yeah, maybe. Its not necessarily that I wanted it to tie into the console definition list, but I just wanted to have a "topic" for each message, but also avoid overcomplicating the data structure.

I have no problem reducing it to just a list of messages. I will do that.